pyteal/ast/box.py

```diff
@@ -0,0 +1,226 @@
+from typing import TYPE_CHECKING
+from pyteal.ast.maybe import MaybeValue
+from pyteal.errors import verifyTealVersion
+
+from pyteal.types import TealType, require_type
+from pyteal.ir import TealOp, Op, TealBlock
+from pyteal.ast.expr import Expr
+
+if TYPE_CHECKING:
+    from pyteal.compiler import CompileOptions
+
+
+class BoxCreate(Expr):
+    """Create a box with a given name and size."""
+
+    def __init__(self, name: Expr, size: Expr) -> None:
+        """
+        Args:
+            name: The key used to reference this box. Must evaluate to a bytes.
+            size: The number of bytes to reserve for this box. Must evaluate to a uint64.
+        """
+
+        super().__init__()
+        require_type(name, TealType.bytes)
+        require_type(size, TealType.uint64)
+        self.name = name
+        self.size = size
+
+    def __teal__(self, options: "CompileOptions"):
+        verifyTealVersion(
+            Op.box_create.min_version,
+            options.version,
+            f"{Op.box_create} unavailable.",
+        )
```

Comment:
@michaeldiamant I noticed that, Jacob is working on consolidating TEAL to AVM version, and I am concerned that, the work of making `verifyTealVersion` (`verifyAVMVersion`) arguments to be keywords, would be interrupted because potential merge conflict.

I would like to wait until Jacob's work merged in, and stuffs are updated in avm8 branch according, then start a new PR for this single purpose, for `verifyTealVersion` appears across the codebase.

Reply:
no worries, no change to implementation.